### PR TITLE
CRUD code style fixes

### PR DESCRIPTION
**Pull request by arogachev**
- Corrected imports to be in alphabetical order.
- Removed extra line breaks and added new.
- When the number of params is 1, it should be "one liner" for more compact view (except very long line). If the user need to add more, he should add line breaks and extra comma on his own.
- Corrected PHPDoc (missing `@inheritdoc`, `more explicit `@return`, missing `@throws`).
- Prevented reaching max symbols limit in line.
- Encoding of title is redundant by default, because this string is safe and managed by programmer, not user.
- Removed extra concatenation with space.
- Converted double quotes to single used in empty string (`''` instead of `""`).
- Capitalized comments.
- Removed variable assignment in condition, I think it's bad practice and it's better to assign variable separately before, this makes code more clear and easier to read.

### CODE REVIEW

File: generators/crud/default/views/create.php [hunk 1]
```diff
@@ -19,12 +16,9 @@
...
 ?>
-<div class="<?= Inflector::camel2id(StringHelper::basename($generator->modelClass)) ?>-create">
 
-    <h1><?= "<?= " ?>Html::encode($this->title) ?></h1>
-
-    <?= "<?= " ?>$this->render('_form', [
-        'model' => $model,
-    ]) ?>
+<div class="<?= Inflector::camel2id(StringHelper::basename($generator->modelClass)) ?>-create">
+    <h1><?= "<?= " ?>$this->title</h1>
```
Comment (githubjeka) [inline generators/crud/default/views/create.php]: ууу куда дел Html::encode ?

File: generators/crud/default/controller.php [hunk 2]
```diff
@@ -72,32 +76,27 @@ public function actionIndex()
             'dataProvider' => $dataProvider,
         ]);
 <?php else: ?>
-        $dataProvider = new ActiveDataProvider([
-            'query' => <?= $modelClass ?>::find(),
-        ]);
+        $dataProvider = new ActiveDataProvider(['query' => <?= $modelClass ?>::find()]);
```
Comment (samdark) [inline generators/crud/default/controller.php]: It was OK as is.

File: generators/crud/default/controller.php [hunk 3]
```diff
@@ -72,32 +76,27 @@ public function actionIndex()
...
 
-        return $this->render('index', [
-            'dataProvider' => $dataProvider,
-        ]);
+        return $this->render('index', ['dataProvider' => $dataProvider]);
```
Comment (samdark) [inline generators/crud/default/controller.php]: It was OK as is.

File: generators/crud/default/controller.php [hunk 4]
```diff
@@ -72,32 +76,27 @@ public function actionIndex()
...
-     * @return mixed
+     * @return string
+     * @throws NotFoundHttpException if the model cannot be found
      */
     public function actionView(<?= $actionParams ?>)
     {
-        return $this->render('view', [
-            'model' => $this->findModel(<?= $actionParams ?>),
-        ]);
+        return $this->render('view', ['model' => $this->findModel(<?= $actionParams ?>)]);
```
Comment (samdark) [inline generators/crud/default/controller.php]: It was OK as is.

File: generators/crud/default/controller.php [hunk 5]
```diff
@@ -72,32 +76,27 @@ public function actionIndex()
...
     }
 
     /**
      * Creates a new <?= $modelClass ?> model.
      * If creation is successful, the browser will be redirected to the 'view' page.
-     * @return mixed
+     * @return string|\yii\web\Response
```
Comment (samdark) [inline generators/crud/default/controller.php]: In this case it's string.

File: generators/crud/default/controller.php [hunk 6]
```diff
@@ -106,9 +105,7 @@ public function actionCreate()
         if ($model->load(Yii::$app->request->post()) && $model->save()) {
             return $this->redirect(['view', <?= $urlParams ?>]);
         } else {
-            return $this->render('create', [
-                'model' => $model,
-            ]);
+            return $this->render('create', ['model' => $model]);
```
Comment (samdark) [inline generators/crud/default/controller.php]: It was OK as is.

File: generators/crud/default/controller.php [hunk 7]
```diff
@@ -116,7 +113,7 @@ public function actionCreate()
      * Updates an existing <?= $modelClass ?> model.
      * If update is successful, the browser will be redirected to the 'view' page.
      * <?= implode("\n     * ", $actionParamComments) . "\n" ?>
-     * @return mixed
+     * @return string|\yii\web\Response
```
Comment (samdark) [inline generators/crud/default/controller.php]: In this case it's string.

File: generators/crud/default/controller.php [hunk 8]
```diff
@@ -125,9 +122,7 @@ public function actionUpdate(<?= $actionParams ?>)
         if ($model->load(Yii::$app->request->post()) && $model->save()) {
             return $this->redirect(['view', <?= $urlParams ?>]);
         } else {
-            return $this->render('update', [
-                'model' => $model,
-            ]);
+            return $this->render('update', ['model' => $model]);
```
Comment (samdark) [inline generators/crud/default/controller.php]: It was OK as is.

File: generators/crud/default/controller.php [hunk 9]
```diff
@@ -135,7 +130,8 @@ public function actionUpdate(<?= $actionParams ?>)
      * Deletes an existing <?= $modelClass ?> model.
      * If deletion is successful, the browser will be redirected to the 'index' page.
      * <?= implode("\n     * ", $actionParamComments) . "\n" ?>
-     * @return mixed
+     * @return \yii\web\Response
```
Comment (samdark) [inline generators/crud/default/controller.php]: In this case it's string.

File: generators/crud/default/search.php [hunk 10]
```diff
@@ -56,30 +56,26 @@ public function scenarios()
...
     {
         $query = <?= isset($modelAlias) ? $modelAlias : $modelClass ?>::find();
 
-        // add conditions that should always apply here
+        // Add conditions that should always apply here
 
-        $dataProvider = new ActiveDataProvider([
-            'query' => $query,
-        ]);
+        $dataProvider = new ActiveDataProvider(['query' => $query]);
```
Comment (samdark) [inline generators/crud/default/search.php]: It was OK as is.

File: generators/crud/default/views/create.php (hunk 1 above)
Comment (samdark) [inline generators/crud/default/views/create.php]: Yep. `Html::encode` removal would lead to XSS.

File: generators/crud/default/views/create.php [hunk 11]
```diff
@@ -19,12 +16,9 @@
...
 
+    <?= "<?= " ?>$this->render('_form', ['model' => $model]) ?>
```
Comment (samdark) [inline generators/crud/default/views/create.php]: It was OK as is.

File: generators/crud/default/views/index.php [hunk 12]
```diff
@@ -22,23 +22,20 @@
 $this->title = <?= $generator->generateString(Inflector::pluralize(Inflector::camel2words(StringHelper::basename($generator->modelClass)))) ?>;
 $this->params['breadcrumbs'][] = $this->title;
 ?>
-<div class="<?= Inflector::camel2id(StringHelper::basename($generator->modelClass)) ?>-index">
 
-    <h1><?= "<?= " ?>Html::encode($this->title) ?></h1>
+<div class="<?= Inflector::camel2id(StringHelper::basename($generator->modelClass)) ?>-index">
+    <h1><?= "<?= " ?>$this->title</h1>
```
Comment (samdark) [inline generators/crud/default/views/index.php]: `Html::encode` again.

File: generators/crud/default/views/index.php [hunk 13]
```diff
@@ -22,23 +22,20 @@
...
 <?php if(!empty($generator->searchModelClass)): ?>
 <?= "    <?php " . ($generator->indexWidgetType === 'grid' ? "// " : "") ?>echo $this->render('_search', ['model' => $searchModel]); ?>
 <?php endif; ?>
 
-    <p>
-        <?= "<?= " ?>Html::a(<?= $generator->generateString('Create ' . Inflector::camel2words(StringHelper::basename($generator->modelClass))) ?>, ['create'], ['class' => 'btn btn-success']) ?>
-    </p>
+    <p><?= "<?= " ?>Html::a(<?= $generator->generateString('Create ' . Inflector::camel2words(StringHelper::basename($generator->modelClass))) ?>, ['create'], ['class' => 'btn btn-success']) ?></p>
```
Comment (samdark) [inline generators/crud/default/views/index.php]: It was OK as is.

File: generators/crud/default/views/update.php [hunk 14]
```diff
@@ -11,22 +11,17 @@
 echo "<?php\n";
 ?>
 
-use yii\helpers\Html;
-
 /* @var $this yii\web\View */
 /* @var $model <?= ltrim($generator->modelClass, '\\') ?> */
 
-$this->title = <?= $generator->generateString('Update {modelClass}: ', ['modelClass' => Inflector::camel2words(StringHelper::basename($generator->modelClass))]) ?> . ' ' . $model-><?= $generator->getNameAttribute() ?>;
+$this->title = <?= $generator->generateString('Update {modelClass}: ', ['modelClass' => Inflector::camel2words(StringHelper::basename($generator->modelClass))]) ?> . $model-><?= $generator->getNameAttribute() ?>;
```
Comment (samdark) [inline generators/crud/default/views/update.php]: Why removing space?

File: generators/crud/default/views/update.php [hunk 15]
```diff
@@ -11,22 +11,17 @@
...
 ?>
-<div class="<?= Inflector::camel2id(StringHelper::basename($generator->modelClass)) ?>-update">
 
-    <h1><?= "<?= " ?>Html::encode($this->title) ?></h1>
-
-    <?= "<?= " ?>$this->render('_form', [
-        'model' => $model,
-    ]) ?>
+<div class="<?= Inflector::camel2id(StringHelper::basename($generator->modelClass)) ?>-update">
+    <h1><?= "<?= " ?>$this->title</h1>
```
Comment (samdark) [inline generators/crud/default/views/update.php]: `Html::encode`

File: generators/crud/default/views/view.php [hunk 16]
```diff
@@ -21,9 +21,9 @@
 $this->params['breadcrumbs'][] = ['label' => <?= $generator->generateString(Inflector::pluralize(Inflector::camel2words(StringHelper::basename($generator->modelClass)))) ?>, 'url' => ['index']];
 $this->params['breadcrumbs'][] = $this->title;
 ?>
-<div class="<?= Inflector::camel2id(StringHelper::basename($generator->modelClass)) ?>-view">
 
-    <h1><?= "<?= " ?>Html::encode($this->title) ?></h1>
+<div class="<?= Inflector::camel2id(StringHelper::basename($generator->modelClass)) ?>-view">
+    <h1><?= "<?= " ?>$this->title</h1>
```
Comment (samdark) [inline generators/crud/default/views/view.php]: `Html::encode`

**Comment by samdark** [comment]
> When the number of params is 1, it should be "one liner" for more compact view (except very long line). If the user need to add more, he should add line breaks and extra comma on his own.

It's more readable as is.

File: generators/crud/default/views/update.php (hunk 14 above)
Comment (arogachev) [inline generators/crud/default/views/update.php]: What's the purpose of it?

**Comment by samdark** [comment]
> Encoding of title is redundant by default, because this string is safe and managed by programmer, not user.

It won't hurt to encode in this case. Using model title as part of the title is very likely.

**Comment by arogachev** [comment]
Disagree about "one liners" Why we need 3 lines instead of 1 when using just one parameter? For more than one, yes, we need separate it to different lines.

**Comment by arogachev** [comment]
I don't understand about `Html::encode()`. Why we need encode strings such as `Create post`. What's unsafe in it? And it's managed only by programmer here. For more complex cases (for example when user input is involved), obviously, user need to add it, but on his own.

**Comment by samdark** [comment]
> Why we need 3 lines instead of 1 when using just one parameter (and when this line is pretty short)?

From my experience, sooner or later, you're adding more arguments for `->render()` calls.

> Why we need encode strings such as Create post. What's unsafe in it?

Nothing until one decides to use post name in the edit form (which is common). Gii is a template so it's not a good idea to provide defaults which lead to insecure solutions.

**Comment by arogachev** [comment]
About adding more arguments - isn't this premature optimization?

**Comment by arogachev** [comment]
I understood you about `Html::encode()`, I think in this case we need to remove it only in `create` and `update` views, because there are no model titles. For `update` and `view ` views we can leave it as it initialliy was.

**Comment by arogachev** [comment]
As for redirects - why it's string? It's more explicit variation than `@mixed`. Because there are only two options, I think it's better explicitly specify them.

**Comment by klimov-paul** [comment]
Explicit specification will be `string|\yii\web\Response`.
As for me this is too long and makes controller code unecessary heavy.
Action methods usually are not invoked directly, so there is no benefit for marking exact result for them.

**Comment by samdark** [comment]
Reviewed again. Won't accept it.

**Pull request closed without merging**